array-reduce/main.js

```diff
@@ -0,0 +1,45 @@
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const account = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traits = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+const add = (previous, current) => previous + current;
+const sum = numbers.reduce(add);
+console.log('sum', sum);
+
+const multi = (previous, current) => previous * current;
+const product = numbers.reduce(multi);
+console.log('product', product);
+
+let total = 0;
+const bank = (previous, current, index) => {
+  if (account[index].type === 'deposit') {
+    total = total + account[index].amount;
+  } else if (account[index].type === 'withdrawal') {
+    total = total - account[index].amount;
+  }
```

Comment:
While this works it would be better to use `current` in place of `account[index]`. They are the same value but you might not always have direct access to the original array plus it is bad practice to use variables defined outside a function within the function.